realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -1826,14 +1823,14 @@ void BaseRealSenseNode::setupStreams()
                     updateStreamCalibData(video_profile);
 
                     // stream index: 1=left, 2=right
-                    if (video_profile.stream_index() == 1) { left_profile = std::make_shared<rs2::video_stream_profile>(video_profile); }
-                    if (video_profile.stream_index() == 2) { right_profile = std::make_shared<rs2::video_stream_profile>(video_profile);  }
+                    if (video_profile.stream_index() == 1) { _left_stereo_profile = std::make_shared<rs2::video_stream_profile>(std::move(video_profile)); }
```

Comment:
The `video_profile` object is created as a temporary stream profile and since we won't be using that beyond line 1826-1827 there is no point in creating a completely new `video_stream_profile` object. Hence we can just move `video_profile` data into the shared object. To summarize, this trick is done to avoid an unnecessary memory allocation which would otherwise be performed by `std::make_shared`.